types/website-scraper/index.d.ts

```diff
@@ -43,8 +43,12 @@ declare namespace websiteScraper {
     interface Callback {
         (error: any | null, result: Resource[] | null): void;
     }
-    function scrape(options: Options, callback: Callback): void;
-    function scrape(options: Options): Promise<Resource[]>;
+    interface scrape {
+        (options: Options, callback: Callback): void;
+        (options: Options): Promise<Resource[]>;
+    }
 }
 
+declare var websiteScraper: websiteScraper.scrape;
```

Comment:
I haven't seen typeof in declaration file yet.
You need to convert functions to the interface, cause, in this case, there won't be duplication. :)